lib/utils/getServerlessConfigFile.js

```diff
@@ -46,28 +53,58 @@ const getServerlessConfigFilePath = serverless => {
   );
 };
 
-const handleJsConfigFile = jsConfigFile =>
+const resolveTsNode = serviceDir => {
+  const resolveModuleRealPath = (...args) =>
+    resolveModulePath(...args).then(({ realPath }) => realPath);
+
+  const createOnError = cb => error => {
+    if (error.code !== 'MODULE_NOT_FOUND') throw error;
+    return cb();
+  };
+
+  const resolveAsServerlessPeerDependency = () => resolveModuleRealPath(__dirname, 'ts-node');
+  const resolveAsServiceDependency = () => resolveModuleRealPath(serviceDir, 'ts-node');
+  const resolveAsGlobalInstallation = () =>
+    spawn('npm', ['root', '-g']).then(({ stdoutBuffer }) => {
+      return require.resolve(`${String(stdoutBuffer).trim()}/ts-node`);
+    });
+  const throwTsNodeError = () => {
+    throw new ServerlessError(
+      'Ensure "ts-node" dependency when working with TypeScript configuration files'
+    );
```

Comment:
For better processing (e.g. in tests), it'll be nice to add _code_ to this error (it's accepted as second argument), e.g. `'TS_NODE_NOT_FOUND'`

Reply:
done and used in test :)